### PR TITLE
#34741 Fix pseudo columns and pseudo tables highlighting

### LINKED ISSUE #34741: Some reserved keywords of sql language use wrong color
### Description

use this code (it has no error but it is colored with "SQL semantic error foreground"):
`SELECT CURRENT_DATE;`

### DBeaver Version

Community Edition 24.1.3.202407042322

### Operating System

Windows 10

### Database and driver

PostgreSQL 16
PostgreSQL JDBC Driver 42.7.2

### PR DESCRIPTION
**Pull request by E1izabeth**
1. Dual table for Oracle.
2. Pseudo columns for Oracle https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Pseudocolumns.html
 - ORA_ROWSCN
 - ROWID
 3. Pseudo columns for SQLite https://www.sqlite.org/lang_createtable.html#rowid Would be nice to also test on old SQLite versions (version < 3.8, 3.8 < version <3.30, version > 3.30)  
  - rowid
  - oid
  - _rowid_
 4. Pseudo columns for PostgreSQL https://www.postgresql.org/docs/8.2/ddl-system-columns.html
  - oid
  - tableoid
  - xmin
  - cmin
  - xmax
  - cmax
  - ctid
  and global variables
  - current_date
  - current_time
  - current_timestamp
  - localtime
  - localtimestamp
  - current_role
  - current_user
  - current_catalog
  - current_schema
  - session_user
  - system_user
  - user
  5. Global variables for DB2
   - CLIENT_IPADDR
   - DATASLICEID
   
   
   Please check pseudo columns visibility in different select query parts. If column is highlighted with red, then it's not available and the query can't run. If there is no red highlighting for pseudo attribute or global variable, then query should run. I mostly tested on plain select and subqueries. If you find that visibility depends on some other conditions, that are not taken in account yet, we can create a new issue for that. Unfortunately, documentation doesn't describe it well.
   Global variables are always visible.

### CODE REVIEW

**Review by ShadelessFox — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/model/DB2DataSource.java [hunk 1]
```diff
@@ -70,7 +73,29 @@
...
+
+    private static final DBDPseudoAttribute[] KNOWN_GLOBAL_PSEUDO_ATTRS = new DBDPseudoAttribute[] {
+        new DBDPseudoAttribute(
+            DBDPseudoAttributeType.OTHER,
+            "CLIENT_IPADDR",
+            null,
+            null,
+            "Contains the value of the client IP address for the connection.",
+            true,
+            DBDPseudoAttribute.PropagationPolicy.GLOBAL_VARIABLE
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.db2/src/org/jkiss/dbeaver/ext/db2/model/DB2DataSource.java]: Make them (GLOBAL_VARIABLEs) to be a list of constants in dialect

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreTable.java [hunk 2]
```diff
@@ -56,6 +60,35 @@ public abstract class PostgreTable extends PostgreTableReal
 {
     private static final Log log = Log.getLog(PostgreTable.class);
 
+    /**
+     * see https://www.postgresql.org/docs/8.2/ddl-system-columns.html
+     * 'oid' described at {@link PostgreConstants#PSEUDO_ATTR_OID}
+     */
+    private static final Map<DBDPseudoAttributeType, List<Pair<String, String>>> ALWAYS_PRESENTED_PSEUDO_ATTR_ENTRIES = Map.of(
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreTable.java]: In fact these attributes are real. We skip them in columns read method. But we can read them from database. Maybe we should (read + hide)?

**Comment by uslss** [comment]
Simple select query check:
1. Dual tables
image
2. Pseudo columns, oracle
image
image
3. Pseudo columns for SQLite
SQLite 3.46.1
image
3.8.6
image
3.28.0
image

4. PG
image
image

5. DB2
image

Also tested in where clause and group by

**Comment by uslss** [comment]
verified

**Pull request merged**